components/style/parser.rs

```diff
@@ -74,3 +74,7 @@ pub fn log_css_error(input: &mut Parser, position: SourcePosition, message: &str
 pub trait Parse {
     fn parse(input: &mut Parser) -> Result<Self, ()> where Self: Sized;
 }
+
+pub trait ParseWithContext {
```

Comment:
What about something like (probably with the traits renamed since we probably want `ParseWithContext` to be the default):

```
pub trait Parse : Sized {
    fn parse(input: &mut Parser) -> Result<Self, ()>;
}

pub trait ParseWithContext : Sized {
    fn parse_with_context(input: &mut Parser, context: &ParserContext) -> Result<Self, ()>;
}

impl<T: Parse> ParseWithContext for T {
    fn parse_with_context(input: &mut Parser, context: &ParserContext) -> Result<Self, ()> {
        T::parse(input)
    }
}
```

In any case agreed, this is kind of offtopic for this PR :)